File: uhrwerk-config/src/main/java/io/qimia/uhrwerk/config/representation/Target.java
package io.qimia.uhrwerk.config.representation;


import io.qimia.uhrwerk.config.ConfigException;

import java.util.Arrays;

public class Target{

    private String connection_name;
    private String format;

    public Target() {}

    public String getConnection_name() {
        return connection_name;
    }

    public void setConnection_name(String connection_name) {
        this.connection_name = connection_name;
    }

    public String getFormat() {
        return format;
    }

    public void setFormat(String format) {
        this.format = format;
    }

    public void validate(String path){
        path += "target/";
        if(connection_name == null){
            throw new ConfigException("Missing field: " + path + "connection_name");
        }
        if(format == null){
            throw new ConfigException("Missing field: " + path + "format");
        }
        if (!Arrays.asList("json", "parquet", "jdbc", "orc", "libsvm", "csv", "text" , "avro").contains(format)) {
            throw new ConfigException("Wrong format! '" + format + "' is not allowed in " + path + "format");
        }
    }

    @Override
    public String toString() {
        return "Target{" +
                "connection_name='" + connection_name + '\'' +
                ", format='" + format + '\'' +
                '}';
    }
}

File: uhrwerk-config/src/main/kotlin/io/qimia/uhrwerk/config/representation/Target.kt
package io.qimia.uhrwerk.config.representation

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty
import io.qimia.uhrwerk.config.builders.ConfigException
import java.util.*

@JsonIgnoreProperties(ignoreUnknown = true)
class Target(
    @JsonProperty("connection_name")
    var connectionName: String? = null,
    var format: String? = null,
    @JsonProperty("table_name")
    var tableName: String? = null,
) {
    fun validate(path: String) {
        var path = path
        path += "target/"
        if (connectionName == null) {
            throw ConfigException("Missing field: " + path + "connection_name")
        }
        if (format == null) {
            throw ConfigException("Missing field: " + path + "format")
        }
        if (!Arrays.asList("json", "parquet", "jdbc", "orc", "libsvm", "csv", "text", "avro", "redshift")
                .contains(format)
        ) {
            throw ConfigException("Wrong format! '" + format + "' is not allowed in " + path + "format")
        }
    }
}